webapi/src/main/scala/org/knora/webapi/responders/v1/ResourcesResponderV1.scala

```diff
@@ -943,23 +943,19 @@ class ResourcesResponderV1 extends Responder {
                     // Filter the source objects by eliminating the ones that the user doesn't have permission to see.
                     sourceObjectsWithPermissions = sourceObjects.filter(sourceObj => sourceObj.permissionCode.nonEmpty)
 
-                    //_ = println(ScalaPrettyPrinter.prettyPrint(sourceObjectsWithPermissions))
-
                     contextItems = sourceObjectsWithPermissions.map {
-                        (sourceObj: SourceObject) =>
+                        sourceObj: SourceObject =>
 
-                            val preview: Option[LocationV1] = sourceObj.fileValues.find(fileVal => fileVal.permissioCode.nonEmpty && fileVal.image.isPreview) match {
-                                case Some(preview: StillImageFileValue) =>
-                                    Some(valueUtilV1.fileValueV12LocationV1(preview.image))
-                                case None => None
-                            }
+                            // Because of #1068, we ignore file values representing preview images. Instead, we generate
+                            // a IIIF preview URL from the full-size image.
 
-                            val locations: Option[Seq[LocationV1]] = sourceObj.fileValues.find(fileVal => fileVal.permissioCode.nonEmpty && !fileVal.image.isPreview) match {
+                            val (preview: Option[LocationV1], locations: Option[Seq[LocationV1]]): (Option[LocationV1], Option[Seq[LocationV1]]) = sourceObj.fileValues.find(fileVal => fileVal.permissionCode.nonEmpty && !fileVal.image.isPreview) match {
```

Comment:
Will this still work if there is no preview image in the triplestore, so the SPARQL query only returns one row per resource?

Reply:
Not sure, will check.

Reply:
Yes, it works: I deleted two preview images from the incunabula test data, and the context test still passes.